uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/paho/MqttConnectionImpl.java

```diff
@@ -297,6 +278,37 @@ private MqttConnectionOptions convertConnectParams(MqttLib.ConnectionParams conn
         return connectionOptions;
     }
 
+    private MqttProperties createPublishProperties(Message message) {
+        MqttProperties properties = new MqttProperties();
+
+        if (message.getUserProperties() != null && !message.getUserProperties().isEmpty()) {
+            List<UserProperty> userProperties = convertToUserProperties(message.getUserProperties());
+            properties.setUserProperties(userProperties);
+            userProperties.forEach(p -> logger.atInfo()
+                    .log("Publish MQTT userProperties: {}, {}", p.getKey(), p.getValue()));
+        }
+
+        String contentType = message.getContentType();
+        if (contentType != null && !contentType.isEmpty()) {
+            properties.setContentType(contentType);
+            logger.atInfo().log("Publish MQTT payload content type '{}'", contentType);
+        }
+
+        Boolean payloadFormatIndicator = message.getPayloadFormatIndicator();
+        if (payloadFormatIndicator != null) {
+            properties.setPayloadFormat(payloadFormatIndicator);
+            logger.atInfo().log("Publish MQTT payload format indicator '{}'", payloadFormatIndicator);
+        }
+
+        Integer messageExpiryInterval = message.getMessageExpiryInterval();
```

Comment:
Use hasMessageFormatIndicator()

Reply:
hasMessageFormatIndicator() is already used in GRPCControlServer